### ISSUE TITLE
How do screen reader users know when they navigate to an accordion widget?

### ISSUE THREAD

**Issue by ianthedev** [w3c/aria-practices#1010]
When screen reader users navigate to a tab widget, they can be informed of that because tab widget has `role="tab"` and `role="tabpanel"`. Therefore they know that they can use the Right Arrow and Left Arrow keys to move focus to the next and previous tabs, respectively.

But accordion widget does not have its exclusive aria roles. How do screen reader users know when they navigate to an accordion widget and know that:

- They can use the Down Arrow and Up Arrow keys to moves focus to the next and previous accordion headers, respectively.
- They can use the Home and End keys to moves focus to the first and last accordion headers, respectively.

----

By the way, the [jQuery UI Accordion Widget](https://jqueryui.com/accordion/) is actually implemented as a tab widget. It adds `role="tab"` on accordion headers and `role="tabpanel"` on accordion panels, and it also wraps accordion headers and accordion panels inside an element which has `role="tablist"`. Is that approach okay?

**Comment by patrickhlauke** [comment]
AT users won't know an accordion is an accordion - it will be exposed as a series of individual disclosure widgets (using the new suggested way of doing things per ARIA practices 1.1).

The down/up arrow/home/end interactions are marked as optional, and indeed won't always work when screen readers are running (taking a random combination of, say, Chrome+NVDA, down/up arrow are the standard reading keys, and NVDA won't pass these on to the page/JavaScript but simply move the reading focus)

The way jQuery does it is the old ARIA practices 1.0 suggested way of doing accordions. Structurally, I'd say it's flawed (as it mixes the tab list and the tab panels, with the latter being awkwardly part of the former). I seem to remember that being a reason why in 1.1 this pattern was abandoned in favour of the more straightforward "series of disclosure widgets" one.

**Comment by Decrepidos** [comment]
@ianthedev FYI functional accessible interaction designs, are included in the AccDC TSG Coding Arena at
https://github.com/accdc/t
Comprehensive explanations are available for relevant control types, including browser and screen reader compatibility testing data as it relates to dynamic content rendering and ARIA, and related pitfalls to be aware of during development.

WhatSock’s The ARIA Role Matrices explains all the ARIA roles and how they should work.  http://whatsock.com/training/matrices/

**Comment by ianthedev** [comment]
@patrickhlauke Thank you for letting me know that how the the accordion widget is designed is totally fine for screen readers. I will implemented it by referring to the existing [W3C example](https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html) then.

@Decrepidos Thank you for WhatSock's information.
